cider/exec/module/batch/CiderBatchUtils.cpp

```diff
@@ -434,20 +435,14 @@ std::string extractUtf8ArrowArrayAt(const ArrowArray* array, size_t index) {
   return std::string(res);
 }
 
-CiderBatch convertToArrowRepresentation(const CiderBatch& output_batch) {
-  std::shared_ptr<CiderTableSchema> table_schema = output_batch.schema();
-  auto column_num = table_schema->getFlattenColCount();
-  auto arrow_colum_num = output_batch.column_num();
-  CHECK_EQ(column_num * 2, arrow_colum_num);
-  auto table_row_num = output_batch.row_num();
-  ArrowArrayBuilder builder;
-  builder = builder.setRowNum(table_row_num);
-  const auto& types = table_schema->getFlattenColumnTypes();
-  const auto& names = table_schema->getFlattenColNames();
-  CHECK_EQ(types.size(), names.size());
-  const int8_t** table_ptr = output_batch.table();
-  for (auto i = 0; i < types.size(); ++i) {
-    const ::substrait::Type& type = types[i];
+int convertToStructArrow(ArrowArrayBuilder& builder,
+                         const ::substrait::Type type,
+                         const int8_t** table_ptr,
+                         int table_ptr_idx,
+                         int table_row_num,
+                         int column_num,
+                         std::vector<std::string> names) {
+  if (!type.has_struct_()) {
```

Comment:
Q: What if we have struct or other nested type?

Reply:
This is a recursive function to build nested type. L445 is the terminate condition, which will build a primitive column and return, L467-L476 will build a struct recursively.